sdk/mobile-center-updates/src/main/java/com/microsoft/azure/mobile/updates/Updates.java

```diff
@@ -374,6 +374,13 @@ private synchronized void cancelPreviousTasks() {
     private synchronized void resumeUpdateWorkflow() {
         if (mForegroundActivity != null && !mWorkflowCompleted && isInstanceEnabled()) {
 
+            /* Don't go any further if the app was installed from an app store. */
```

Comment:
I meant you can move this to the top of the method (above the if statement).
Basically we already know that the application cannot be updated at the time Updates service is initialized. You can check this early.